Edge 92 release 1.0.902.49 and Edge 93 prerelease 1.0.955-prerelease (#1359)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/release-notes.md

> [!NOTE]
> WebView2 bug fixes are either Runtime or SDK specific.  

## 1.0.955-prerelease

> Release Date: July 27th, 2021  

Reviewer: ```suggestion
Release Date: July 26, 2021  
```

[NuGet package][NuGetGallery1.0.955-prerelease] \| Minimum Microsoft Edge version to load: 86.0.616.0 or newer \| Full API Compatibility: 93.0.955.0 or newer  

### General  
*   General WebView2 performance improvements.
*   Added some ETW tracing support for WebView2.
*   Removed Microsoft branding from edge://history.
*   WebView2 has a new default Download UI.

> #### Experimental Features  

Reviewer 2: One more feature I collected is Disable PDF Tool Bar(https://microsoft.visualstudio.com/DefaultCollection/Edge/_git/chromium.src/pullrequest/6255496?path=%2Fedge_embedded_browser%2Fclient%2Fwin%2Fcurrent%2Fwebview2staging.idl), it that expected to not be included here? I'm not sure, just confirm.

Reviewer: @jasonstephen15 here's the link
```
[Webview2ReferenceWin32Icorewebview2experimentalsettings6ViewWebview210955PrereleaseGetHiddenpdftoolbaritems]: /microsoft-edge/webview2/reference/win32/icorewebview2experimentalsettings6?view=webview2-1.0.955-prerelease&preserve-view=true#get_hiddenpdftoolbaritems "get_HiddenPdfToolbarItems - interface ICoreWebView2ExperimentalSettings6 | Microsoft Docs"
```

*   Added [OpenTaskManagerWindow API][Webview2ReferenceWin32Icorewebview2experimental4ViewWebview210955PrereleaseOpentaskmanagerwindow] to launch a webview2 browser task manager.
*   Added [NewWindowRequestedEventArgs][Webview2ReferenceWin32Icorewebview2experimentalnewwindowrequestedeventargsViewWebview210955PrereleaseGetName].

#### Bug fixes  

> *   Fixed bug that broke the edge://downloads and edge://history pages. This change is Runtime-specific.

Reviewer 3: Do we need to include this one here?
Work item: https://microsoft.visualstudio.com/Edge/_workitems/edit/32155177
PR: https://microsoft.visualstudio.com/DefaultCollection/Edge/_git/chromium.src/pullrequest/6095899

A short description for this would be: “Added support for virtual host name mapping to work with Service Worker.”.

*   Fixed bugs to improve reliability in the WebView2Loader.dll.

> *   Fixed bug in which new window request handler launches two window when handling links with target=_blank.

Reviewer: ```suggestion
*   Fixed bug in which NewWindowRequested event handler launches two window when handling links with target=_blank.
```

*   Fixed a bug in WebView visual hosting would present a black flicker before startup.
*   Fixed bug when add_WebResourceRequested was not working for WebView2 created via add_NewWindowRequested. \([\#616][GithubMicrosoftedgeWebviewfeedbackIssue616]\)

> *   Fixed a bug where closing out a child form while the find in page bar is open crashes webview2 in parent form. \([\#1097][GithubMicrosoftedgeWebviewfeedbackIssue1097]\)

Reviewer: Should this be under .NET?

*   Fixed a bug where closing out a child form while the find in page bar is open crashes webview2 in parent form. \([\#1097][GithubMicrosoftedgeWebviewfeedbackIssue1097]\)

## 1.0.902.49

Release Date: July 26, 2021  

[NuGet package][NuGetGallery1.0.902.49] \| Minimum Runtime version to load: 86.0.616.0 or newer \| Full API Compatibility: 92.0.902.49 or newer  

### General  

#### Bug Fixes  
*   Fix bug that broke IsBuiltInErrorPageEnabled property to disable built in error page for navigation failure and render process failure.  This change is Runtime-specific. \([\#634][GithubMicrosoftedgeWebviewfeedbackIssue634]\)
*   Fixed an issue where the WebView2 would steal focus away from the user's focus.
*   Fixed bug when AddScriptToExecuteOnDocumentCreated was not working on child windows. \([\#935][GithubMicrosoftedgeWebviewfeedbackIssue935]\)
*   Fixed bug which caused inactive tabs to be auto discarded. \([\#637][GithubMicrosoftedgeWebviewfeedbackIssue637]\)
*   Fixed a bug that some times the Navigation Id of NavigationCompleted Event can be wrong if the navigation is interrupted by another navigation. \([\#1142][GithubMicrosoftedgeWebviewfeedbackIssue1142]\)

> #### Promotions

Reviewer 2: It seems the Download API is stable as well now, are we missing it here or that is expected?

Author: @peiche-jessica do you know why this did not show up in the script? It should be stable yes. Same with AddHostObjectToScriptWithOrigins. I will add it for now.

Reviewer: @jasonstephen15 I think I meant to bundle the FrameCreated event and all ICoreWebView2Frame ones (including ICoreWebView2Frame::AddHostObjectToScriptWithOrigins) into one bullet point.

> [!NOTE]
> WebView2 bug fixes are either Runtime or SDK specific.  

## 1.0.955-prerelease

Release Date: July 26, 2021  

> [NuGet package][NuGetGallery1.0.955-prerelease] \| Minimum Microsoft Edge version to load: 86.0.616.0 or newer \| Full API Compatibility: 93.0.955.0 or newer  

Reviewer: Can we make a one-time change to Full API Compatibility to 93.0.967.0 for 1.0.955-prerelease? This is for a bug fixing usage of `ICoreWebView2_5::add_ClientCertificatedRequested` that @vickiez just checked into main.

```suggestion
[NuGet package][NuGetGallery1.0.955-prerelease] \| Minimum Microsoft Edge version to load: 86.0.616.0 or newer \| Full API Compatibility: 93.0.967.0 or newer  
```

### General  
*   General WebView2 performance improvements.
*   Added some ETW tracing support for WebView2.
*   Removed Microsoft branding from edge://history.
*   WebView2 has a new default Download UI.

#### Experimental Features  
*   Added [OpenTaskManagerWindow API][Webview2ReferenceWin32Icorewebview2experimental4ViewWebview210955PrereleaseOpentaskmanagerwindow] to launch a webview2 browser task manager.
*   Added [NewWindowRequestedEventArgs][Webview2ReferenceWin32Icorewebview2experimentalnewwindowrequestedeventargsViewWebview210955PrereleaseGetName].

> *   Added [PDF Toolbar API][Webview2ReferenceWin32Icorewebview2experimentalnewwindowrequestedeventargsViewWebview210955PrereleaseGetName].

Author: I am not sure the proper link, commented out for now, we can make a pr later to add that in.

*   Fixed a bug in WebView visual hosting would present a black flicker before startup.
*   Fixed bug when add_WebResourceRequested was not working for WebView2 created via add_NewWindowRequested. \([\#616][GithubMicrosoftedgeWebviewfeedbackIssue616]\)
*   Allowed the host app to set foreground on a different application in response to certain events (NavigationStarting, AddHostObjectToScript methods, WebMessageReceived, and NewWindowRequested). \([\#1092][GithubMicrosoftedgeWebviewfeedbackIssue1092]\)
*   Fix bug to fire PermissionRequested event for Microphone. This change is Runtime-specific.\([\#1462][GithubMicrosoftedgeWebviewfeedbackIssue1462]\)
*   Fixed bug when ExecuteScriptAsync was blocking forever after several successful runs. This change is Runtime-specific. \([\#1348][GithubMicrosoftedgeWebviewfeedbackIssue1348]\)
*   Fixed bug that prevented non-ASCII filenames from being used for `ResultFilePath` in DownloadStartingEventArgs. \([\#1428][GithubMicrosoftedgeWebviewfeedbackIssue1428]\)
*   Fixed bug where the default popup's title bar is not displayed entirely. This change is Runtime-specific. \([\#1016][GithubMicrosoftedgeWebviewfeedbackIssue1016]\)

> *   Added support for virtual host name mapping to work with Service Workers.

Reviewer 4: This sounds like a new feature. Do we need a new section heading for that?

Author: changed to be in another section
